### PR TITLE
AML emission

### PR DESCRIPTION
**Pull request by nschejtman**
Refactors: split clases into multiple files
Added Dialect Instances AML rendering test suite

### CODE REVIEW

File: shared/src/main/scala/amf/plugins/document/vocabularies/emitters/instances/DeclarationsGroupEmitter.scala [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+                                    declarationsPath: Seq[String],
+                                    aliases: Map[String, (String, String)],
+                                    keyPropertyId: Option[String] = None,
+                                    renderOptions: RenderOptions)
+    extends EntryEmitter
+    with DialectEmitterHelper {
+
+  def computeIdentifier(decl: DialectDomainElement): String = {
+    decl.declarationName.option() match {
+      case Some(name) => name
```
Comment (pope1838) [inline shared/src/main/scala/amf/plugins/document/vocabularies/emitters/instances/DeclarationsGroupEmitter.scala:27]: Field that semantically represents name? Like schema-org#name? or http://a.ml/vocabularies/amf/core##Name, shacl#Name (in a first version maybe anything that has a name "Name" : .#Name?)

File: shared/src/main/scala/amf/plugins/document/vocabularies/emitters/instances/DialectEmitterHelper.scala [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+      })
+    }
+    else {
+      Nil
+    }
+  }
+
+  def findNodeMappingById(nodeMappingId: String): (Dialect, NodeMappable) = {
+    maybeFindNodeMappingById(nodeMappingId).getOrElse {
+      throw new Exception(s"Cannot find node mapping $nodeMappingId")
```
Comment (pope1838) [inline shared/src/main/scala/amf/plugins/document/vocabularies/emitters/instances/DialectEmitterHelper.scala:48]: If not mapping is present, maybe we can skip this node and don't stop all the emitton?
Later we could add an Error Handler to report error as AMF does.

File: shared/src/main/scala/amf/plugins/document/vocabularies/emitters/instances/DeclarationsGroupEmitter.scala [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+    }
+
+    if (declarationsPath.isEmpty) {
+      val declarationKey = publicNodeMapping.name().value()
+      b.entry(
+          declarationKey,
+          _.obj { b =>
+            sortedDeclarations().foreach { decl =>
+              val identifier = computeIdentifier(decl)
+              b.entry(
```
Comment (pope1838) [inline shared/src/main/scala/amf/plugins/document/vocabularies/emitters/instances/DeclarationsGroupEmitter.scala:52]: If we extract the entry emittion to another emitter of a NamedDialectDomainElement we can map the declaration to emitter and do the traverse+sorted as amf wbapi emitter does.
Also, that will be like a first step to create a declared element isolated emiter.

File: shared/src/main/scala/amf/plugins/document/vocabularies/emitters/instances/DeclarationsGroupEmitter.scala (hunk 1 above)
Comment (nschejtman) [reply shared/src/main/scala/amf/plugins/document/vocabularies/emitters/instances/DeclarationsGroupEmitter.scala:27]: Good suggestion, keeping it in mind for further refactors.

File: shared/src/main/scala/amf/plugins/document/vocabularies/emitters/instances/DialectEmitterHelper.scala (hunk 2 above)
Comment (nschejtman) [reply shared/src/main/scala/amf/plugins/document/vocabularies/emitters/instances/DialectEmitterHelper.scala:48]: Good suggestion, keeping it in mind for further refactors.

File: shared/src/main/scala/amf/plugins/document/vocabularies/emitters/instances/DeclarationsGroupEmitter.scala (hunk 3 above)
Comment (nschejtman) [reply shared/src/main/scala/amf/plugins/document/vocabularies/emitters/instances/DeclarationsGroupEmitter.scala:52]: Good suggestion, keeping it in mind for further refactors.

**Review by pope1838 — APPROVED** [review]

**Pull request merged**